File: Debug/Python/9.py
# To find the highest priority in a queue based on the lowest number of running time

import heapq

class Task:
    def __init__(self, priority, running_time):
        self.priority = priority
        self.running_time = running_time
    def __lt__(self, other):
        return (self.running_time, self.priority) < (other.running_time, other.priority)

def find_highest_priority(tasks):
    min_heap = {}
    for task in tasks:
        heapq.heappush(task,min_heap)
    highest_priority_task = heapq.heappush(min_heap)
    return highest_priority_task

tasks = [Task(1, 5), Task(2, 3), Task(3, 7), Task(1, 4)]
highest_priority_task = find_highest_priority(tasks)
print("Highest priority task: Priority -", highest_priority_task.priority, ", Running Time -", highest_priority_task.running_time)

File: Debug/C++/9.cpp
// To find the highest priority in a queue based on the lowest number of running time

#include <iostream>
#include <queue>
#include <vector>

using namespace std;

class Task
{
public:
    int priority;
    int runningTime;

    Task(int priority, int runningTime) : priority(priority), runningTime(runningTime) {}

    bool operator<(const Task &other) const
    {
        if (this->runningTime != other.runningTime)
        {
            return this->priority < other.priority;
        }
        else
        {
            return this->runningTime < other.runningTime;
        }
    }
};

Task findHighestPriority(vector<Task> &tasks)
{
    priority_queue<Task> minHeap;
    for (Task &task : tasks)
    {
        minHeap.push(task);
    }
    Task highestPriorityTask = minHeap.top();
    minHeap.pop();
    return highestPriorityTask;
}

int main()
{
    vector<Task> tasks = {Task(1, 5), Task(2, 3), Task(3, 7), Task(1, 4)};
    Task highestPriorityTask = findHighestPriority(tasks);
    cout << "Highest priority task: Priority - " << highestPriorityTask.priority << ", Running Time - " << highestPriorityTask.runningTime << endl;
    return 0;
}
